upstairs/src/lib.rs

```diff
@@ -1784,6 +1772,31 @@ impl std::ops::Deref for Buffer {
     }
 }
 
+/// Buffer with unknown contents, which must be reset before it can be used
+#[derive(Debug, Default)]
+pub(crate) struct UninitializedBuffer(Buffer);
+
+impl UninitializedBuffer {
+    /// Resets to the given size, clearing the `owned` array
+    ///
+    /// The data array is resized (padding with 0s) but is not cleared; it is
+    /// expected that this function will be called right before reading into it,
+    /// so clearing it is unnecessary.
+    pub(crate) fn reset_owned(
```

Comment:
naming nit: You could just call this `reset`, considering how it's leaving practically nothing about the buffer untouched

Reply:
I wanted to make it distinct from `Buffer::reset`, which sets all data values to zero; this function leaves data values (below the resizing point) set to their previous values.

Reply:
Fair enough.